src/arch/x86/smbios.c

```diff
@@ -594,6 +594,68 @@ static int smbios_write_type11(unsigned long *current, int *handle)
 	return len;
 }
 
+static int smbios_write_type16(unsigned long *current, int *handle)
+{
+	// read SPD info from file
```

Comment:
@finus16 this has to be board specific. Everything implemented here should be in mainboard directory. Everything what You implemented here should land in a `fill_mainboard_smbios_type16` which would be called in `smbios_write_type16`. `smbios_write_type16` should call `fill_mainboard_smbios_type16` and return the length, update current and handle pointers

Reply:
@miczyg1 in such way?

Reply:
@finus16 yes, something like that